src/effect.cpp

```diff
@@ -446,11 +446,12 @@ std::string effect::disp_name() const
 
     // End result should look like "name (l. arm)" or "name [intensity] (l. arm)"
     std::ostringstream ret;
-    if (eff_type->use_name_ints()) {
-        if(eff_type->name[intensity - 1] == "") {
+    if( eff_type->use_name_ints() ) {
+        const std::string &d_name = eff_type->name[ std::min<size_t>( intensity, eff_type->name.size() ) - 1 ];
```

Comment:
0 intensity would cause other bugs. I'm pretty sure it is set to 1 even if explicitly requested to be 0.
`name` can't be empty - `use_name_ints` means `name.size() > 1`.